core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -240,4 +330,87 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public static class InternalSearchResponse {
```

Comment:
I think this is still needed as its own class because it is used as an "incomplete" search response in e.g. SeachPhaseController#merge() and in some other places where some information we need for the SearchResponse ctor is not available yet. I also wanted to keep the amount of changes small, so I'd prefer to leave it for now. This would be another refactoring that is not necessary for parsing.